packages/migrations/src/migration.ts

```diff
@@ -16,7 +17,7 @@ import { erc20TokenInfo, erc721TokenInfo } from './utils/token_info';
  * @param txDefaults Default transaction values to use when deploying contracts (e.g., specify the desired contract creator with the `from` parameter).
  * @returns The addresses of the contracts that were deployed.
  */
-export async function runMigrationsAsync(provider: Provider, txDefaults: TxData): Promise<ContractAddresses> {
+export async function runMigrationsAsync(provider: Web3ProviderEngine, txDefaults: TxData): Promise<ContractAddresses> {
```

Comment:
We use web3 provider engine internally but this is also a public package. I don't think we need to require this to be a Web3ProviderEngine do we?